mujoco_warp/_src/constraint.py

```diff
@@ -481,6 +480,69 @@ def _efc_friction(
   )
 
 
+@wp.kernel
+def _efc_friction_tendon(
+  # Model:
+  nv: int,
+  opt_timestep: float,
+  tendon_solref_fri: wp.array(dtype=wp.vec2),
+  tendon_solimp_fri: wp.array(dtype=vec5),
+  tendon_frictionloss: wp.array(dtype=float),
+  tendon_invweight0: wp.array(dtype=float),
+  # Data in:
+  qvel_in: wp.array2d(dtype=float),
+  ten_J_in: wp.array3d(dtype=float),
+  # In:
+  refsafe_in: int,
+  # Data out:
+  nf_out: wp.array(dtype=int),
+  nefc_out: wp.array(dtype=int),
+  efc_worldid_out: wp.array(dtype=int),
+  efc_id_out: wp.array(dtype=int),
+  efc_J_out: wp.array2d(dtype=float),
+  efc_pos_out: wp.array(dtype=float),
+  efc_margin_out: wp.array(dtype=float),
+  efc_D_out: wp.array(dtype=float),
+  efc_aref_out: wp.array(dtype=float),
+  efc_frictionloss_out: wp.array(dtype=float),
+):
+  worldid, tenid = wp.tid()
+
+  if tendon_frictionloss[tenid] <= 0.0:
+    return
+
+  efcid = wp.atomic_add(nefc_out, 0, 1)
+  wp.atomic_add(nf_out, 0, 1)
+  efc_worldid_out[efcid] = worldid
+
+  Jqvel = float(0.0)
+  for i in range(nv):
```

Comment:
let's put a TODO to figure out how to parallelize this

Reply:
done